packages/components/src/menu-item/stories/index.story.tsx

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import type { Meta, StoryFn } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { link, more, check } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import MenuGroup from '../../menu-group';
+import MenuItem, { MenuItem as UnforwardedMenuItem } from '..';
+import Shortcut from '../../shortcut';
+
+const meta: Meta< typeof UnforwardedMenuItem > = {
+	component: UnforwardedMenuItem,
+	title: 'Components/MenuItem',
+	argTypes: {
+		children: { control: { type: null } },
+		icon: {
+			control: { type: 'select' },
+			options: [ 'check', 'link', 'more' ],
+			mapping: {
+				check,
+				link,
+				more,
+			},
+		},
+	},
+	parameters: {
+		controls: {
+			expanded: true,
+			include: [
+				'className',
+				'children',
+				'info',
+				'icon',
+				'iconPosition',
+				'isSelected',
+				'role',
+				'suffix',
+				'label',
+			],
+		},
+		docs: { source: { state: 'open' } },
+	},
+};
+export default meta;
+
+const Template: StoryFn< typeof MenuItem > = ( props ) => {
+	return (
+		<MenuGroup>
+			<MenuItem { ...props }>Menu Item 1</MenuItem>
+		</MenuGroup>
+	);
+};
+
+export const Default: StoryFn< typeof MenuItem > = Template.bind( {} );
+
+export const IsSelected = Template.bind( {} );
+IsSelected.args = {
+	...Default.args,
+	isSelected: true,
+};
+
+export const WithIcon = Template.bind( {} );
+WithIcon.args = {
+	...Default.args,
+	icon: check,
```

Comment:
Nit: I would prefer using an icon other than `check` here, because it looks too similar to the UI pattern for `MenuItemsChoice`.

```suggestion
	icon: link,
```

Reply:
Sorted
<img width="756" alt="image" src="https://github.com/WordPress/gutenberg/assets/10071857/dad47e36-88da-46aa-9cdd-5dd2a1ae03c8">